src/Structure/Content/MetaBox.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * Abstract class for meta box registration.
+ *
+ * @author  Tom McFarlin <tom.mcfarlin@webdevstudios.com>
+ * @package WebDevStudios\OopsWP\Structure\Content
+ * @since   2019-05-01
+ */
+
+namespace WebDevStudios\OopsWP\Structure\Content;
+
+use Exception;
+use WebDevStudios\OopsWP\Utility\Registerable;
+
+/**
+ * Class MetaBox
+ *
+ * @author  Tom McFarlin <tom.mcfarlin@webdevstudios.com>
+ * @package WebDevStudios\OopsWP\Structure\Content
+ * @since   2019-05-01
+ */
+abstract class MetaBox extends ContentType implements Registerable {
+
+	/**
+	 * The slug used to identify this meta box.
+	 *
+	 * @var   string
+	 * @since 2019-05-01
+	 */
+	protected $slug;
+
+	/**
+	 * The title displayed at the top of the meta box.
+	 *
+	 * @var   string
+	 * @since 2019-05-01
+	 */
+	protected $title;
+
+	/**
+	 * The types of post in which this meta box should be displayed.
+	 *
+	 * @var   array
+	 * @since 2019-05-01
+	 */
+	protected $post_types;
```

Comment:
The [Developer Resources](https://developer.wordpress.org/reference/functions/add_meta_box/) outlines this as:

> Accepts a single screen ID, WP_Screen object, or array of screen IDs. Default is the current screen

And I _usually_ use `$post_types` here rather than a specific screen object or ID. To make this as flexible as possible, though, I can refactor it to accept any of the above types.